### DISCUSSION TITLE
Launching after install?

### DISCUSSION THREAD

**Discussion by kalkal11** [hlky/stable-diffusion#106] [Q&A] [answered]
I'm going to take one for the team in asking what is probably an obvious answer to many.

What is the correct process of launching this after the first install? I just keep running the webui.cmd file, which I'm almost certain is not correct. (however works fine)

Which python file am I launching?

I assume `scripts\webui.py` but that causes an error.

Also, how do I get a gradio link? Currently I just get the local URL to use.

Thanks so much!

**Comment by GalaxyTimeMachine** [comment]
I believe that running the webui.cmd file is the correct start method. That will then check for any changes in the environment.yaml and keep it up-to-date if it changes.

To get an external Gradio link you would normally add " --share" to the end of the start command, however I just opened an issue as that doesn't work with the webui.cmd script. A workaround would be to edit the webui.py file and set the "default=False" to "True" on the end of the line (25) for `parser.add_document("--share"`...
That will require a restart of the server and it should then list the external address below the local one.

Don't forget to set it back to "False" when you're done, as anyone can access it remotely. There is also an option to set a password, but I haven't got that to work properly yet.

**Comment by kalkal11** [comment]
![image](https://user-images.githubusercontent.com/110500565/187296688-2ed678d0-bcf6-468d-a6e4-7d685f982e98.png)
thing is I get this when trying to launch from webui.py, it works with the cmd file though, strange

**Comment by GalaxyTimeMachine** [comment, marked as answer]
You need to use webui.cmd to start it. I had a response and it seems the way to enable the share option is to add it to the relauncher.py file.
eg. Add it to this line:

`os.system("python scripts/webui.py --share")`

If you also want to add a password, then add it like this:

`os.system("python scripts/webui.py --share --share-password xxxxxxxx")`

**Comment by jpbonadio** [comment]
I was able to make it work changing a line in the webui.py file.
At around line 1610 (more or less because I've made some other changes) you need to change the line that launches the demo to this:
self.demo.launch(share=True)

Then remove the --share on the launch cell. It will be like this:
%cd /content/stable-diffusion
!python /content/stable-diffusion/scripts/webui.py \
--ckpt '{models_path}/sd-v1-4.ckpt' \
--outdir '{output_path}' \
 {vars}

**Reply by GalaxyTimeMachine** [reply]
My solution above works and is the recommended method. It's a much smaller file, so easier to find the required line to edit.
